packages/graphql/tests/tck/directives/authorization/projection.test.ts

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { gql } from "graphql-tag";
+import type { DocumentNode } from "graphql";
+import { Neo4jGraphQL } from "../../../../src";
+import { createJwtRequest } from "../../../utils/create-jwt-request";
+import { formatCypher, translateQuery, formatParams } from "../../utils/tck-test-utils";
+
+describe("Cypher Auth Projection", () => {
+    const secret = "secret";
+    let typeDefs: DocumentNode;
+    let neoSchema: Neo4jGraphQL;
+
+    beforeAll(() => {
+        typeDefs = gql`
+            type User {
+                id: ID
+                name: String
+            }
+
+            extend type User {
+                id: ID @authorization(validate: [{ when: BEFORE, where: { node: { id: "$jwt.sub" } } }])
+            }
+        `;
+
+        neoSchema = new Neo4jGraphQL({
+            typeDefs,
+            features: {
+                authorization: {
+                    key: secret,
+                },
+            },
+        });
+    });
+
+    test("Update Node", async () => {
+        const query = gql`
+            mutation {
+                updateUsers(update: { id: "new-id" }) {
+                    users {
+                        id
+                    }
+                }
+            }
+        `;
+
+        const req = createJwtRequest("secret", { sub: "super_admin", roles: ["admin"] });
+        const result = await translateQuery(neoSchema, query, {
+            req,
+        });
+
+        expect(formatCypher(result.cypher)).toMatchInlineSnapshot(`
+            "MATCH (this:\`User\`)
```

Comment:
This snapshot looks as though it is validating before and after. However, the type defs seem to suggest that it should only validate before

Reply:
Ignore me, I realise it's validating for both the READ and UPDATE operations